runtime/tests/checker/arrays_dictionaries_test.go

```diff
@@ -1225,6 +1225,104 @@ func TestCheckResourceArrayFilterInvalid(t *testing.T) {
 	assert.IsType(t, &sema.InvalidResourceArrayMemberError{}, errs[0])
 }
 
+func TestCheckArrayMap(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+		fun test() {
+			let x = [1, 2, 3]
+			let trueForEven =
+				fun (_ x: Int): Bool {
+					return x % 2 == 0
+				}
+
+			let y = x.map(trueForEven)
```

Comment:
For all the checker tests, maybe declare the `y` variable with explicit type, so the test will also check whether the return type of the map function in each case is correct. Currently, it is not being tested/asserted.

```suggestion
			let y: [Int] = x.map(trueForEven)
```